configuration.md
---
# !!! DO NOT MODIFY !!!
# This file is auto-generated by the gendoc tool based on the source code of the app.
description: This section describes the configuration parameters and their types for INX-Faucet.
keywords:
- IOTA Node
- Hornet Node
- Faucet
- Configuration
- JSON
- Customize
- Config
- reference
---


# Core Configuration

INX-Faucet uses a JSON standard format as a config file. If you are unsure about JSON syntax, you can find more information in the [official JSON specs](https://www.json.org).

You can change the path of the config file by using the `-c` or `--config` argument while executing `inx-faucet` executable.

For example:
```bash
inx-faucet -c config_defaults.json
```

You can always get the most up-to-date description of the config parameters by running:

```bash
inx-faucet -h --full
```
## <a id="app"></a> 1. Application

| Name                      | Description                                            | Type    | Default value |
| ------------------------- | ------------------------------------------------------ | ------- | ------------- |
| checkForUpdates           | Whether to check for updates of the application or not | boolean | true          |
| [shutdown](#app_shutdown) | Configuration for shutdown                             | object  |               |

### <a id="app_shutdown"></a> Shutdown

| Name                     | Description                                                                                            | Type   | Default value |
| ------------------------ | ------------------------------------------------------------------------------------------------------ | ------ | ------------- |
| stopGracePeriod          | The maximum time to wait for background processes to finish during shutdown before terminating the app | string | "5m"          |
| [log](#app_shutdown_log) | Configuration for log                                                                                  | object |               |

### <a id="app_shutdown_log"></a> Log

| Name     | Description                                         | Type    | Default value  |
| -------- | --------------------------------------------------- | ------- | -------------- |
| enabled  | Whether to store self-shutdown events to a log file | boolean | true           |
| filePath | The file path to the self-shutdown log              | string  | "shutdown.log" |

Example:

```json
  {
    "app": {
      "checkForUpdates": true,
      "shutdown": {
        "stopGracePeriod": "5m",
        "log": {
          "enabled": true,
          "filePath": "shutdown.log"
        }
      }
    }
  }
```

## <a id="logger"></a> 2. Logger

| Name        | Description                                                                                                    | Type   | Default value |
| ----------- | -------------------------------------------------------------------------------------------------------------- | ------ | ------------- |
| name        | The optional name of the logger instance. All log messages are prefixed with that name.                        | string | ""            |
| level       | The minimum enabled logging level                                                                              | string | "info"        |
| timeFormat  | Sets the logger's timestamp format. (options: "rfc3339", "rfc3339nano", "datetime", "timeonly", and "iso8601") | string | "rfc3339"     |
| outputPaths | A list of file paths or stdout/stderr to write logging output to                                               | array  | stdout        |

Example:

```json
  {
    "logger": {
      "name": "",
      "level": "info",
      "timeFormat": "rfc3339",
      "outputPaths": [
        "stdout"
      ]
    }
  }
```

## <a id="inx"></a> 3. INX

| Name                  | Description                                                                                        | Type   | Default value    |
| --------------------- | -------------------------------------------------------------------------------------------------- | ------ | ---------------- |
| address               | The INX address to which to connect to                                                             | string | "localhost:9029" |
| maxConnectionAttempts | The amount of times the connection to INX will be attempted before it fails (1 attempt per second) | uint   | 30               |
| targetNetworkName     | The network name on which the node should operate on (optional)                                    | string | ""               |

Example:

```json
  {
    "inx": {
      "address": "localhost:9029",
      "maxConnectionAttempts": 30,
      "targetNetworkName": ""
    }
  }
```

## <a id="faucet"></a> 4. Faucet

| Name                           | Description                                                                                                                  | Type    | Default value    |
| ------------------------------ | ---------------------------------------------------------------------------------------------------------------------------- | ------- | ---------------- |
| baseTokenAmount                | The amount of funds the requester receives                                                                                   | uint    | 1000000000       |
| baseTokenAmountSmall           | The amount of funds the requester receives if the target address has more funds than the faucet amount and less than maximum | uint    | 100000000        |
| baseTokenAmountMaxTarget       | The maximum allowed amount of funds on the target address                                                                    | uint    | 5000000000       |
| manaAmount                     | The amount of mana the requester receives                                                                                    | uint    | 1000000          |
| manaAmountMinFaucet            | The minimum amount of mana the faucet needs to hold before mana payouts become active                                        | uint    | 1000000000       |
| tagMessage                     | The faucet transaction tag payload                                                                                           | string  | "FAUCET"         |
| batchTimeout                   | The maximum duration for collecting faucet batches                                                                           | string  | "2s"             |
| bindAddress                    | The bind address on which the faucet website can be accessed from                                                            | string  | "localhost:8091" |
| [rateLimit](#faucet_ratelimit) | Configuration for rateLimit                                                                                                  | object  |                  |
| [pow](#faucet_pow)             | Configuration for pow                                                                                                        | object  |                  |
| debugRequestLoggerEnabled      | Whether the debug logging for requests should be enabled                                                                     | boolean | false            |

### <a id="faucet_ratelimit"></a> RateLimit

| Name        | Description                                     | Type    | Default value |
| ----------- | ----------------------------------------------- | ------- | ------------- |
| enabled     | Whether the rate limiting should be enabled     | boolean | true          |
| period      | The period for rate limiting                    | string  | "5m"          |
| maxRequests | The maximum number of requests per period       | int     | 10            |
| maxBurst    | Additional requests allowed in the burst period | int     | 20            |

### <a id="faucet_pow"></a> Pow

| Name        | Description                                                                              | Type | Default value |
| ----------- | ---------------------------------------------------------------------------------------- | ---- | ------------- |
| workerCount | The amount of workers used for calculating PoW when sending payloads to the block issuer | int  | 4             |

Example:

```json
  {
    "faucet": {
      "baseTokenAmount": 1000000000,
      "baseTokenAmountSmall": 100000000,
      "baseTokenAmountMaxTarget": 5000000000,
      "manaAmount": 1000000,
      "manaAmountMinFaucet": 1000000000,
      "tagMessage": "FAUCET",
      "batchTimeout": "2s",
      "bindAddress": "localhost:8091",
      "rateLimit": {
        "enabled": true,
        "period": "5m",
        "maxRequests": 10,
        "maxBurst": 20
      },
      "pow": {
        "workerCount": 4
      },
      "debugRequestLoggerEnabled": false
    }
  }
```

## <a id="profiling"></a> 5. Profiling

| Name        | Description                                       | Type    | Default value    |
| ----------- | ------------------------------------------------- | ------- | ---------------- |
| enabled     | Whether the profiling component is enabled        | boolean | false            |
| bindAddress | The bind address on which the profiler listens on | string  | "localhost:6060" |

Example:

```json
  {
    "profiling": {
      "enabled": false,
      "bindAddress": "localhost:6060"
    }
  }
```



documentation/docs/configuration.md
---
# !!! DO NOT MODIFY !!!
# This file is auto-generated by the gendoc tool based on the source code of the app.
description: This section describes the configuration parameters and their types for INX-Faucet.
keywords:
- IOTA Node
- Hornet Node
- Faucet
- Configuration
- JSON
- Customize
- Config
- reference
---


# Core Configuration

INX-Faucet uses a JSON standard format as a config file. If you are unsure about JSON syntax, you can find more information in the [official JSON specs](https://www.json.org).

You can change the path of the config file by using the `-c` or `--config` argument while executing `inx-faucet` executable.

For example:
```bash
inx-faucet -c config_defaults.json
```

You can always get the most up-to-date description of the config parameters by running:

```bash
inx-faucet -h --full
```
## <a id="app"></a> 1. Application

| Name                      | Description                                            | Type    | Default value |
| ------------------------- | ------------------------------------------------------ | ------- | ------------- |
| checkForUpdates           | Whether to check for updates of the application or not | boolean | true          |
| [shutdown](#app_shutdown) | Configuration for shutdown                             | object  |               |

### <a id="app_shutdown"></a> Shutdown

| Name                     | Description                                                                                            | Type   | Default value |
| ------------------------ | ------------------------------------------------------------------------------------------------------ | ------ | ------------- |
| stopGracePeriod          | The maximum time to wait for background processes to finish during shutdown before terminating the app | string | "5m"          |
| [log](#app_shutdown_log) | Configuration for log                                                                                  | object |               |

### <a id="app_shutdown_log"></a> Log

| Name     | Description                                         | Type    | Default value  |
| -------- | --------------------------------------------------- | ------- | -------------- |
| enabled  | Whether to store self-shutdown events to a log file | boolean | true           |
| filePath | The file path to the self-shutdown log              | string  | "shutdown.log" |

Example:

```json
  {
    "app": {
      "checkForUpdates": true,
      "shutdown": {
        "stopGracePeriod": "5m",
        "log": {
          "enabled": true,
          "filePath": "shutdown.log"
        }
      }
    }
  }
```

## <a id="logger"></a> 2. Logger

| Name              | Description                                                                 | Type    | Default value |
| ----------------- | --------------------------------------------------------------------------- | ------- | ------------- |
| level             | The minimum enabled logging level                                           | string  | "info"        |
| disableCaller     | Stops annotating logs with the calling function's file name and line number | boolean | true          |
| disableStacktrace | Disables automatic stacktrace capturing                                     | boolean | false         |
| stacktraceLevel   | The level stacktraces are captured and above                                | string  | "panic"       |
| encoding          | The logger's encoding (options: "json", "console")                          | string  | "console"     |
| outputPaths       | A list of URLs, file paths or stdout/stderr to write logging output to      | array   | stdout        |
| disableEvents     | Prevents log messages from being raced as events                            | boolean | true          |

Example:

```json
  {
    "logger": {
      "level": "info",
      "disableCaller": true,
      "disableStacktrace": false,
      "stacktraceLevel": "panic",
      "encoding": "console",
      "outputPaths": [
        "stdout"
      ],
      "disableEvents": true
    }
  }
```

## <a id="inx"></a> 3. INX

| Name                  | Description                                                                                        | Type   | Default value    |
| --------------------- | -------------------------------------------------------------------------------------------------- | ------ | ---------------- |
| address               | The INX address to which to connect to                                                             | string | "localhost:9029" |
| maxConnectionAttempts | The amount of times the connection to INX will be attempted before it fails (1 attempt per second) | uint   | 30               |
| targetNetworkName     | The network name on which the node should operate on (optional)                                    | string | ""               |

Example:

```json
  {
    "inx": {
      "address": "localhost:9029",
      "maxConnectionAttempts": 30,
      "targetNetworkName": ""
    }
  }
```

## <a id="faucet"></a> 4. Faucet

| Name                           | Description                                                                                                                  | Type    | Default value    |
| ------------------------------ | ---------------------------------------------------------------------------------------------------------------------------- | ------- | ---------------- |
| amount                         | The amount of funds the requester receives                                                                                   | uint    | 1000000000       |
| smallAmount                    | The amount of funds the requester receives if the target address has more funds than the faucet amount and less than maximum | uint    | 100000000        |
| maxAddressBalance              | The maximum allowed amount of funds on the target address                                                                    | uint    | 2000000000       |
| maxOutputCount                 | The maximum output count per faucet message                                                                                  | int     | 128              |
| tagMessage                     | The faucet transaction tag payload                                                                                           | string  | "HORNET FAUCET"  |
| batchTimeout                   | The maximum duration for collecting faucet batches                                                                           | string  | "2s"             |
| bindAddress                    | The bind address on which the faucet website can be accessed from                                                            | string  | "localhost:8091" |
| [rateLimit](#faucet_ratelimit) | Configuration for rateLimit                                                                                                  | object  |                  |
| debugRequestLoggerEnabled      | Whether the debug logging for requests should be enabled                                                                     | boolean | false            |

### <a id="faucet_ratelimit"></a> RateLimit

| Name        | Description                                     | Type    | Default value |
| ----------- | ----------------------------------------------- | ------- | ------------- |
| enabled     | Whether the rate limiting should be enabled     | boolean | true          |
| period      | The period for rate limiting                    | string  | "5m"          |
| maxRequests | The maximum number of requests per period       | int     | 10            |
| maxBurst    | Additional requests allowed in the burst period | int     | 20            |

Example:

```json
  {
    "faucet": {
      "amount": 1000000000,
      "smallAmount": 100000000,
      "maxAddressBalance": 2000000000,
      "maxOutputCount": 128,
      "tagMessage": "HORNET FAUCET",
      "batchTimeout": "2s",
      "bindAddress": "localhost:8091",
      "rateLimit": {
        "enabled": true,
        "period": "5m",
        "maxRequests": 10,
        "maxBurst": 20
      },
      "debugRequestLoggerEnabled": false
    }
  }
```

## <a id="profiling"></a> 5. Profiling

| Name        | Description                                       | Type    | Default value    |
| ----------- | ------------------------------------------------- | ------- | ---------------- |
| enabled     | Whether the profiling plugin is enabled           | boolean | false            |
| bindAddress | The bind address on which the profiler listens on | string  | "localhost:6060" |

Example:

```json
  {
    "profiling": {
      "enabled": false,
      "bindAddress": "localhost:6060"
    }
  }
```



components/faucet/params.go
package faucet

import (
	"time"

	"github.com/iotaledger/hive.go/app"
)

type ParametersFaucet struct {
	BaseTokenAmount          uint64        `default:"1000000000" usage:"the amount of funds the requester receives"`
	BaseTokenAmountSmall     uint64        `default:"100000000" usage:"the amount of funds the requester receives if the target address has more funds than the faucet amount and less than maximum"`
	BaseTokenAmountMaxTarget uint64        `default:"5000000000" usage:"the maximum allowed amount of funds on the target address"`
	ManaAmount               uint64        `default:"1000000" usage:"the amount of mana the requester receives"`
	ManaAmountMinFaucet      uint64        `default:"1000000000" usage:"the minimum amount of mana the faucet needs to hold before mana payouts become active"`
	TagMessage               string        `default:"FAUCET" usage:"the faucet transaction tag payload"`
	BatchTimeout             time.Duration `default:"2s" usage:"the maximum duration for collecting faucet batches"`
	BindAddress              string        `default:"localhost:8091" usage:"the bind address on which the faucet website can be accessed from"`
	RateLimit                struct {
		Enabled     bool          `default:"true" usage:"whether the rate limiting should be enabled"`
		Period      time.Duration `default:"5m" usage:"the period for rate limiting"`
		MaxRequests int           `default:"10" usage:"the maximum number of requests per period"`
		MaxBurst    int           `default:"20" usage:"additional requests allowed in the burst period"`
	}
	PoW struct {
		// the amount of workers used for calculating PoW when sending payloads to the block issuer
		WorkerCount int `default:"4" usage:"the amount of workers used for calculating PoW when sending payloads to the block issuer"`
	} `name:"pow"`
	// DebugRequestLoggerEnabled defines whether the debug logging for requests should be enabled
	DebugRequestLoggerEnabled bool `default:"false" usage:"whether the debug logging for requests should be enabled"`
}

var ParamsFaucet = &ParametersFaucet{}

var params = &app.ComponentParams{
	Params: map[string]any{
		"faucet": ParamsFaucet,
	},
	Masked: nil,
}
